src/blob.rs

```diff
@@ -17,11 +17,56 @@ struct Header {
     flags: u64,
 }
 
-impl<T> Blob<T> where T: Default {
+impl<K> Blob<K> where K: Default {
+    /// # Description
+    /// Create new blob from file
+    // @TODO more useful result
     pub fn from_file<P: AsRef<Path>>(_path: P) -> Result<Self, ()> {
         // @TODO implement
         Ok(Self::default())
     }
+
+    /// # Description
+    /// Writes given slice to file, returns number of written bytes
+    // @TODO more useful result
+    pub fn write(&mut self, _key: &K, _value: &[u8]) -> Result<usize, ()>  {
```

Comment:
We should write some struct and not separate key and value (value is bad name for this any way).

Reply:
IMHO, `key` and `value` should be passed as separate arguments, but I agree that it is better to make `value` a structure.

Reply:
Updated: `write()` takes `Record` struct, which contains key

Reply:
@ikopylov let go with record. to avoid data duplication